lib/hash/hasher.go

```diff
@@ -0,0 +1,17 @@
+package hash
+
+import (
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func ComputeHash(input *unstructured.Unstructured) (string, error) {
+	spec, err := json.Marshal(input.Object["spec"])
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%x", md5.Sum(spec)), nil
+}
```

Comment:
The approach LGTM, one interesting this is that something similar happens in k8s controllers under the hood to generate a [pod-template-hash](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template-hash-label), here's how they do it 
https://github.com/kubernetes/kubernetes/blob/ba10ee7671099254db18f9fe9e779ef6339e935b/pkg/controller/controller_utils.go#L1198-L1213

not sure if we need to mirror the complexity they added, annoyingly we can't use the function since it only accepts a pod spec not a generic map 🙃

Reply:
yeah, i look at it (that's why the code I wrote is very similarly named) and i just can't grasp why so complicated; my feeling is that pod hash collision may be more likely (lots of pods, lots of pods that have the same spec). Went for the simpler approach (md5) until we learn why that's bad

Reply:
> can't grasp why so complicated

same. Sounds good